_posts/2017/2017-03-14-one-development-environment-to-rule-them-all.md

```diff
@@ -0,0 +1,72 @@
+---
+layout: post
+title: "One development environment to rule them all"
+date: "2017-03-14 07:00:00"
+image: blog-header/ring.jpg
+categories: tech-stories
+authors: ["Jordi"]
+---
+
+A usual problem developers are facing, is to work with a common development environment.
+At the same time, this environment has to be the same as the production environment.
+ePages software is working into servers with [centOS](https://www.centos.org/) or [debian](https://www.debian.org/) which makes it mandatory for the developers to write the code in this environment.
+For sure, it is pointless to force all developers to have one of those OS installed, taking into account that the preferred OS are [macOS, ubuntu or windows](https://developer.epages.com/blog/2016/12/22/the-battle-of-the-operating-systems.html).
+
+This scenario calls for finding a solution.
+There are several ways to solve this, and I'll try to explain the different options that we have and which one has mainly been chosen by our development team in Barcelona.
+
+## Convert your laptop into a VM container
+
+One of the easiest ways to find a solution - and also the most popular one - is to use a VM like [vmware](http://www.vmware.com/) or [virtual box](https://www.virtualbox.org/) and add the maximum resources of you laptop to it.
+Then, this VM becomes the main OS of your laptop and all the tools (IDE, scripts, ...) are hosted inside.
+It means, that in fact the laptop is hosting two operative systems but mainly only one is used.
+
+## Containerize the development environment
+
+In order to do a step forward regarding the last proposal and try to not waste resources, the best option is to have the basic environment containerized.
+This means that everybody can get this container and work with the same environment.
+
+This container only needs the basic stuff.
+This way, we can use the tools hosted in our main OS and avoid the waist of resources since it is only the development environment running the code.
+
+This proposal presents one big problem.
```

Comment:
But the challenge is: we need to find... (combine these two sentences; and challenge is nicer than problem ;-))